Guide: Remotely configure RUM using LaunchDarkly (#15983)
DataDog/documentation: content/en/real_user_monitoring/guide/remotely-configure-rum-using-launchdarkly.md

---
title: Remotely configure RUM using LaunchDarkly
kind: guide
beta: false
private: true
description: Learn how to set up RUM with LaunchDarkly to remotely configure RUM sampling.
aliases:
- /real_user_monitoring/guide/remote-config-launchdarkly/
further_reading:
- link: '/real_user_monitoring/explorer'
  tag: 'Documentation'
  text: 'Visualize your RUM data in the RUM Explorer'
---

## Overview
When instrumenting a [RUM application][1], you can remotely control the RUM initialization configurations based on your immediate needs, such as an ongoing incident where you might need higher fidelity data.

Instead of having to deploy the changes to your RUM initialization configurations, you can use feature flags. Feature flag management companies like [LaunchDarkly][2] evaluate feature flags on the server side and thus allow you to make changes to your code without needing to redeploy it. 

## Setting up your flag in LaunchDarkly

> To set up your flag in LaunchDarkly, start by following their documentation on [setting up an SDK][3]. More information on the Client-side SDK is available in LaunchDarkly's [documentation][4].

Reviewer: ```suggestion
To set up your flag in LaunchDarkly, start by following their documentation on [setting up an SDK][3]. For additional details, see LaunchDarkly's [Client-side SDK documentation][4].
```

In this example, you can add an evaluation in JS like the code snippet below.

```javascript
const RUM_sample_rate = client.variation('rum-sample-rate-configuration', false);
```
Then add this to your RUM initialization:

```javascript
datadogRum.init({
  applicationId: '<DATADOG_APPLICATION_ID>',
  clientToken: '<DATADOG_CLIENT_TOKEN>',
  site: '<DATADOG_SITE>',
  service: 'my-web-application',
  env: 'production',
  version: '1.0.0',
  sampleRate: RUM_sample_rate,
  sessionReplaySampleRate: 100,
  trackResources: true,
  trackLongTasks: true,
  trackInteractions: true,
})
```

### Entire RUM configuration option

Before you initialize the RUM SDK, you need to first evaluate your LaunchDarkly feature flags. For example, in JS you can add an evaluation like this:

```javascript
const RUM_configuration = client.variation('rum-configuration', false);
```

However, before you can pass in the configuration to initialize RUM, you'll need to create an Object from the flag JSON value. Once you've done that, you can initialize the RUM SDK.

```javascript
datadogRum.init(RUM_configuration_object)
```

## Embed LaunchDarkly's controls to configure RUM directly in your Dashboards

> If you want to change your RUM configuration directly in your Datadog application, you can embed the LaunchDarkly UI into Datadog and switch your feature flag on/off. The feature flags are set up so you can keep them off, with the default values. When you want to have higher fidelity data, you can turn on your feature flag and the values you set for the ON variation will now be used for the RUM initialization. 

Reviewer: ```suggestion
If you want to change your RUM configuration directly in your Datadog application, you can embed the LaunchDarkly UI into Datadog and switch your feature flag on/off. The feature flags are set up so you can keep them off, with the default values. When you want to have higher fidelity data, you can turn on your feature flag and the values you set for the ON variation will be used for the RUM initialization. 
```
